pallets/asset/src/checkpoint/benchmarking.rs

```diff
@@ -0,0 +1,94 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use crate::benchmarking::owned_ticker;
+use frame_benchmarking::benchmarks;
+use frame_system::RawOrigin;
+use polymesh_primitives::calendar::CalendarUnit;
+
+const MAX_STORED_SCHEDULES: u32 = 1000;
+
+const SCHEDULE: ScheduleSpec = ScheduleSpec {
+    // Not entirely clear this is more expensive, but probably is.
+    // In either case, this triggers `start == now`, which is definitely more expensive.
+    start: None,
+    // Forces us into the most amount of branches.
+    remaining: 2,
+    period: CalendarPeriod {
+        // We want the schedule to be recurring as it triggers more branches.
+        amount: 5,
+        // Months have the most complicated "next at" calculation, so we use months.
+        unit: CalendarUnit::Month,
+    },
+};
+
+fn init<T: Trait>() -> (RawOrigin<T::AccountId>, Ticker) {
+    <pallet_timestamp::Now<T>>::set(1000.into());
+    let (owner, ticker) = owned_ticker::<T>();
+    (owner.origin(), ticker)
+}
+
+fn init_with_existing<T: Trait>(existing: u32) -> (RawOrigin<T::AccountId>, Ticker) {
+    let (owner, ticker) = init::<T>();
+
+    // Ensure max compexity admits `existing + 1` schedules.
+    let max = SCHEDULE.period.complexity() * (existing as u64 + 1);
+    Module::<T>::set_schedules_max_complexity(RawOrigin::Root.into(), max).unwrap();
+
+    // First create some schedules. To make sorting more expensive, we'll make em all equal.
+    for _ in 0..existing {
+        Module::<T>::create_schedule(owner.clone().into(), ticker, SCHEDULE).unwrap();
+    }
+
+    (owner, ticker)
+}
+
+benchmarks! {
+    _ {}
+
+    set_schedules_max_complexity {}: _(RawOrigin::Root, 7)
+    verify {
+        assert_eq!(Module::<T>::schedules_max_complexity(), 7)
+    }
+
+    create_checkpoint {
+        let (owner, ticker) = init::<T>();
+    }: _(owner, ticker)
+    verify {
+        assert_eq!(Module::<T>::checkpoint_id_sequence(ticker), CheckpointId(1))
+    }
+
+    create_schedule {
+        // Stored schedules before creating this one.
+        let s in 0 .. MAX_STORED_SCHEDULES;
```

Comment:
Instead of making this a variable, Let's base it on `SchedulesMaxComplexity`.
1. Create `SchedulesMaxComplexity` number of schedules with 1 complexity each. Just inject them into the `Schedules` storage for our usecase. This will cause the worse case.
2. Set `SchedulesMaxComplexity` to 2*`SchedulesMaxComplexity`, create a new schedule with as complex path as possible while still being under `SchedulesMaxComplexity`.

Reply:
Let's have the `ScheduleMaxComplexity` as the runtime constant, you can provide the declaration in the checkpoint trait definition and use that something like this 

```
let s in 0 .. T::ScheduleMaxComplexity::get() as u32;
```

So its weight will be directly proportional to the `ScheduleMaxComplexity` and when we want to change it we can just change the runtime configuration instead of the weight formula.